modules/e2ee/OlmAdapter.js

```diff
@@ -62,39 +62,57 @@ export class OlmAdapter extends Listenable {
         this._key = undefined;
         this._keyIndex = -1;
         this._reqs = new Map();
+        this._sessionInitialization = undefined;
 
         if (OlmAdapter.isSupported()) {
             this._bootstrapOlm();
 
             this._conf.on(JitsiConferenceEvents.ENDPOINT_MESSAGE_RECEIVED, this._onEndpointMessageReceived.bind(this));
-            this._conf.on(JitsiConferenceEvents.CONFERENCE_JOINED, this._onConferenceJoined.bind(this));
             this._conf.on(JitsiConferenceEvents.CONFERENCE_LEFT, this._onConferenceLeft.bind(this));
+            this._conf.on(JitsiConferenceEvents.USER_JOINED, this._onParticipantJoined.bind(this));
             this._conf.on(JitsiConferenceEvents.USER_LEFT, this._onParticipantLeft.bind(this));
         } else {
             this._init.reject(new Error('Olm not supported'));
         }
     }
 
     /**
-     * Indicates if olm is supported on the current platform.
-     *
-     * @returns {boolean}
+     * Starts new olm sessions with every other participant that has the participantId "smaller" the localParticipantId.
      */
-    static isSupported() {
-        return typeof window.Olm !== 'undefined';
+    async initSessions() {
+        if (this._sessionInitialization) {
+            await this._sessionInitialization;
+        } else {
+            this._sessionInitialization = new Deferred();
+
+            await this._init;
+
+            const promises = [];
+            const localParticipantId = this._conf.myUserId().toString();
+
+            for (const participant of this._conf.getParticipants()) {
+                if (localParticipantId < participant.getId().toString()) {
+                    promises.push(this._sendSessionInit(participant));
+                }
+            }
+
+            await Promise.allSettled(promises);
+
+            // TODO: retry failed ones.
+            // TODO: skip participants which don't support E2EE.
+
+            this._sessionInitialization.resolve();
+            this._sessionInitialization = undefined;
```

Comment:
If you set it to undefined you'll keep trying to establish sessions with all participants once a new one joins AFAIS.

Reply:
I have changed the logic here and onParticipantJoined we just initialise the session with that participant if the condition is met